hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java

```diff
@@ -890,4 +896,35 @@ private void sendResponse(HttpPostCommand command, String value) {
     public void handleRejection(HttpPostCommand command) {
         handle(command);
     }
+
+    private void handleSetLicense(HttpPostCommand command) {
+        final int retryCount = 100;
+        String res;
+        byte[] data = command.getData();
+        try {
+            String[] strList = bytesToString(data).split("&");
+            if (authenticate(command, strList[0], strList.length > 1 ? strList[1] : null)) {
+                // assumes that both groupName and password are present
+                String licenseKey = strList.length > 2 ? URLDecoder.decode(strList[2], "UTF-8") : null;
+                invokeOnStableClusterSerial(textCommandService.getNode().nodeEngine, new Supplier<Operation>() {
```

Comment:
Smells like a nice λ

Reply:
It indeed does, but alas we can't use lambdas in 3.12 (as @kwart reminded me during his review).

Reply:
Oops sorry i thought this was the 4.0 changes